ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/AlterTableHandler.java

```diff
@@ -168,10 +205,19 @@ public void handle(Context withinContext) throws Exception {
       }
     }
 
-    // If the tables are filtered based on name, then needs to handle the rename scenarios.
-    if (!withinContext.replScope.includeAllTables()) {
-      if (!handleForTableLevelReplication(withinContext)) {
-        LOG.info("Alter event for table " + before.getTableName() + " is skipped from dumping");
+    // Handle renames for table level replication.
+    if ((Scenario.RENAME == scenario)
+            && (!withinContext.replScope.includeAllTables() || withinContext.oldReplScope != null)) {
+      String oldName = before.getTableName();
+      String newName = after.getTableName();
+      boolean needDump;
+      if (withinContext.oldReplScope == null) {
+        needDump = handleRenameForTableLevelReplication(withinContext, oldName, newName);
+      } else {
+        needDump = handleRenameForReplacePolicy(withinContext, oldName, newName);
+      }
+      if (!needDump) {
+        LOG.info("Rename event for table " + oldName + " is skipped from dumping");
```

Comment:
Shall change the log to include "newName" as well. ... oldName to newName ...

Reply:
done